File: OS/2do Parcial/Project/Modeling/Project/src/com/data/List.java
package com.data;

public class List<E> {
    private Node<E> front;
    private Node<E> rear;

    int size;

    public List() {
        this.front = null;
        this.rear = null;
        this.size = 0;
    }

    public boolean insert(Node<E> node){
        return true;
    }

    public boolean remove(){
        return true;
    }

    public Node<E> front(){
        return null;
    }

    public Node<E> rear() {
        return rear;
    }

    public int size() {
        return size;
    }

    public boolean isEmpty(){
        return size == 0;
    }

    @Override
    public String toString() {
        String list = "";
        list += "[";
        for (Node aux = front; aux != null; aux = aux.getNext())
        {
            list += (" " + aux.getDatum());
        }
        list += "]";
        return list;
    }
}

File: OS/2do Parcial/Project/Planning Procces/data/list.cpp
#include "list.hpp"

template <typename T>
List<T>::List()
{
    front = nullptr;
    rear = nullptr;
    size = 0;
}

template <typename T>
List<T>::~List()
{
    delete;
}

template <typename T>
bool List<T>::insert(Node<T> datum)
{
    return false;
}

template <typename T>
bool List<T>::remove()
{
    return false;
}

template <typename T>
Node<T> List<T>::getFront()
{
    return nullptr;
}

template <typename T>
Node<T> List<T>::getRear()
{
    return nullptr;
}

template <typename T>
string List<T>::toString(){
    return "";
}